Sources/ReactiveArray.swift

```diff
@@ -0,0 +1,137 @@
+import Foundation
+import ReactiveSwift
+import enum Result.NoError
+
+public enum Change<T> {
+    case insert(element: T, at: Int)
+    case remove(element: T, at: Int)
+}
+
+public final class ReactiveArray<Element> {
+
+    fileprivate var elements: ContiguousArray<Element>
+
+    fileprivate let innerObserver: Observer<[Change<Element>], NoError>
+
+    public let signal: Signal<[Change<Element>], NoError>
+
+    public var capacity: Int {
```

Comment:
The reservation is just a performance hint. It is not a capacity cap, and the contract does not care whether the backing storage does actually reserve the capacity.